File: primatis-api/src/main/java/be/primatis/user/web/CreateUserResponse.java
package be.primatis.user.web;

/**
 * Réponse de {@code POST /api/v1/users} (DEV-05.5). {@code initialPassword}
 * n'est retourné qu'ici, une seule fois : aucun {@code GET} ni {@link
 * UserResponse} ultérieur ne l'expose, et il n'est jamais persisté en clair
 * (seul {@code passwordHash} est stocké sur {@code AppUser}).
 */
public record CreateUserResponse(UserResponse user, String initialPassword) {
}

File: primatis-web/src/app/user/models/create-user-response.ts
import { UserResponse } from './user-response';

/**
 * Voir `be.primatis.user.web.CreateUserResponse` côté backend.
 * `initialPassword` n'est retourné qu'ici, une seule fois.
 */
export interface CreateUserResponse {
  user: UserResponse;
  initialPassword: string;
}
